src/main/scala/org/apache/daffodil/DaffodilPlugin.scala

```diff
@@ -279,7 +279,12 @@ object DaffodilPlugin extends AutoPlugin {
      */
     libraryDependencies ++= {
       if (daffodilBuildsCharset.value) {
-        Seq("org.apache.daffodil" %% "daffodil-io" % daffodilVersion.value % "provided")
+        val charsetDep = Map(
+          ">=4.0.0 " -> "org.apache.daffodil" % "daffodil-core_3" % daffodilVersion.value % "provided",
+          "=3.11.0 " -> "org.apache.daffodil" % "daffodil-io_2.13" % daffodilVersion.value % "provided",
+          "<=3.10.0 " -> "org.apache.daffodil" % "daffodil-io_2.12" % daffodilVersion.value % "provided"
+        )
+        filterVersions(daffodilVersion.value, charsetDep)
```

Comment:
We don't technically need to hardcode the cross version in these dependencies since these libraryDependencies can use `scalaVersion`. So this could be simplified to something like this with %% using the scalaVersion, which gets set appropriate based on daffodilVersion.

```scala
        val charsetDep = Map(
          ">=4.0.0 " -> "org.apache.daffodil" %% "daffodil-core" % daffodilVersion.value % "provided",
          "<=3.11.0" -> "org.apache.daffodil" %% "daffodil-io" % daffodilVersion.value % "provided"
        )
        filterVersions(daffodilVersion.value, charsetDep)
```

Same idea for daffodilBuildsLayer.